pyopencl/array.py

```diff
@@ -2604,11 +2604,20 @@ def make_func_for_chunk_size(chunk_size):
 
 def concatenate(arrays, axis=0, queue=None, allocator=None):
     """
+    Return a :class:`Array` that is a concatenation of the input tuple of 
+    :class:`Array` along :arg axis:. **Warning** Only axis = 0 has been 
+    implemented.
+
     .. versionadded:: 2013.1
     """
     # {{{ find properties of result array
 
     shape = None
+    if axis != 0:
```

Comment:
This isn't correct. This should instead be based on contiguity. Column-major arrays for example allow concatenation along the last axis.

Reply:
I noticed that the code failed the tests because I missed a syntax error. Sorry for the noise. 

About contiguity, I did some tests, and the function does not work in any axis for colum-major arrays.[ In line 2571 of the Array.concatenate function](https://github.com/inducer/pyopencl/blob/dfc4e840cfa4b371a90c624e44bee5399b057681/pyopencl/array.py#L2638), the output array `result` is always a row-major array, leading to the same `NotImlementedError` due to miss-matching strides.

It shouldn't be hard to allow column-major arrays, the issue I see is that the Array class has no member that indicates the ordering. One could deduce the order by looking at the strides (if Array.strides[0] > Array.strides[-1]: row-major, else: column-major) but maybe it is too hacky. what do you think?

Reply:
Thanks for pointing out the issue with `concatenate` and column-major arrays, we should look into fixing that. I agree with your assessment about inspection of the strides, it's not quite elegant.

FWIW, CI is still not passing on this branch.

Reply:
Oh yeah! I forgot to push. It should work now.